internal/nfs/controller/volume.go

```diff
@@ -251,18 +254,22 @@ func (nv *NFSVolume) deleteExportCommand(cmd, nfsCluster string) []string {
 // getNFSCluster fetches the NFS-cluster name from the CephFS journal.
 func (nv *NFSVolume) getNFSCluster() (string, error) {
 	if !nv.connected {
-		return "", fmt.Errorf("can not get NFS-cluster for %q: not connected", nv)
+		return "", fmt.Errorf("can not get NFS-cluster for %q: %w", nv, ErrNotConnected)
 	}
 
 	fs := fscore.NewFileSystem(nv.conn)
 	fsName, err := fs.GetFsName(nv.ctx, nv.fscID)
 	if err != nil {
-		return "", fmt.Errorf("failed to get filesystem name for ID %x: %w", nv.fscID, err)
+		if errors.Is(err, util.ErrPoolNotFound) {
+			return "", fmt.Errorf("%w for ID %x: %v", ErrFilesystemNotFound, nv.fscID, err)
+		}
+
+		return "", fmt.Errorf("filesystem name for ID %x %w: %v", nv.fscID, ErrNotFound, err)
 	}
 
 	mdPool, err := fs.GetMetadataPool(nv.ctx, fsName)
 	if err != nil {
-		return "", fmt.Errorf("failed to get metadata pool for %q: %w", fsName, err)
+		return "", fmt.Errorf("metadata pool for %q %w: %v", fsName, ErrNotFound, err)
```

Comment:
Do we need to return `ErrNotFound` for all the error cases or return `ErrNotFound` only if `ErrPoolNotFound` error is returned?

Reply:
As far as I understood, for now, we are returning this for all cases, and in case we don't find the pool or in the above case the filesystem we can be specific about it. @nixpanic please correct me if I am wrong.

Reply:
I do not have a strong preference for one approach or another. It is true that an error fetching the metadata pool now will assume that the NFS-volume does not exist. Even in the case when the error is different from `ErrPoolNotFound`.

But, I also agree that it is cleaner to do the same checking as done for GetFsName() on line 261. In case the Pool does can not be found, return a `ErrNotFound` and otherwise the `err` that was returned from the call (with additional `fmt.Errorf(..)` context).

The same theory applies to the error checking further below.